jerry-core/ecma/base/ecma-helpers.c

```diff
@@ -861,28 +866,19 @@ void
 ecma_free_property (ecma_object_t *object_p, /**< object the property belongs to */
                     ecma_property_t *prop_p) /**< property */
 {
-  switch ((ecma_property_type_t) prop_p->type)
-  {
-    case ECMA_PROPERTY_NAMEDDATA:
-    {
-      ecma_free_named_data_property (object_p, prop_p);
-
-      break;
-    }
+  JERRY_ASSERT (object_p != NULL && prop_p != NULL);
 
-    case ECMA_PROPERTY_NAMEDACCESSOR:
-    {
-      ecma_free_named_accessor_property (object_p, prop_p);
-
-      break;
-    }
-
-    case ECMA_PROPERTY_INTERNAL:
-    {
-      ecma_free_internal_property (prop_p);
-
-      break;
-    }
+  if (prop_p->flags & ECMA_PROPERTY_FLAG_NAMEDDATA)
+  {
+    ecma_free_named_data_property (object_p, prop_p);
+  }
+  else if (prop_p->flags & ECMA_PROPERTY_FLAG_NAMEDACCESSOR)
+  {
+    ecma_free_named_accessor_property (object_p, prop_p);
+  }
+  else
+  {
```

Comment:
True. But still, IMHO it is good coding practice which helps maintainibility to state that "normally, I'd have written `else if` here but I'm sure that its predicate will always hold therefore I just go with an `else`, however, I still assert on that predicate". And what's more, it does not cost anything in release builds.